src/js/components/mapWidgets/widgetContent/SubscriptionContent.tsx

```diff
@@ -80,14 +62,57 @@ const SubscriptionContent: FunctionComponent = () => {
     //     colorTheme = '#929292';
     // }
 
+    const updateDataset = (): void => {
+      const jsonData: JSONData = {
+        datasets: [],
+        language: subscription.attributes.language,
+        resource: subscription.attributes.resource,
+        params: subscription.attributes.params
+      };
+      const datasetOn = subscription.attributes.datasets.includes(dataset);
+
+      if (datasetOn) {
+        jsonData.datasets = subscription.attributes.datasets.filter(
+          datasetID => datasetID !== dataset
+        );
+      } else {
+        subscription.attributes.datasets.push(dataset);
+        jsonData.datasets = subscription.attributes.datasets;
+      }
+
+      fetch(
+        `https://production-api.globalforestwatch.org/v1/subscriptions/${subscription.id}`,
+        {
+          method: 'PATCH',
+          credentials: 'include',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(jsonData)
+        }
+      )
+        .then(response => (response.status === 200 ? response.json() : null))
+        .then(results => {
+          if (results) {
+            const index = userSubscriptions.findIndex(
+              u => u.id === results.data.id
+            );
+            userSubscriptions[index] = { ...results.data };
```

Comment:
can we make a copy of userSubscriptions instead of modifying it in place? userSubscriptions is coming from redux and what I understand they always say do not modify state directly. thoughts?